src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -0,0 +1,539 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Providers.ImportCompletion;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal static partial class ExtensionMethodImportCompletionHelper
+    {
+        private class ExtensionMethodSymbolComputer
+        {
+            private int Position { get; }
+            private Document CurrentDocument { get; }
+            private SemanticModel SemanticModel { get; }
+            private Project CurrentProject => CurrentDocument.Project;
+            private Solution Solution => CurrentProject.Solution;
+            private ITypeSymbol ReceiverTypeSymbol { get; }
+            private ImmutableArray<string> ReceiverTypeNames { get; }
+            private ISet<string> NamespaceInScope { get; }
+            private ConcurrentDictionary<ITypeSymbol, bool> CheckedReceiverTypes { get; }
+            private IImportCompletionCacheService<CacheEntry, object> CacheService { get; }
```

Comment:
consider making most of these just fields.  for the actual properties, move below the fields.